src/views/Gallery/index.js

```diff
@@ -1,39 +1,35 @@
 import Api from '../../utils/Api';
 import { useEffect, useState } from 'react';
-import SingleImage from '../../shared/components/SingleImage';
+import { useParams } from 'react-router-dom'; 
 import { Title, Wrapper } from './Style';
-import { Link } from 'react-router-dom';
+import ImageGrid from '../../shared/components/ImageGrid';
+
 
 const Gallery = ({ galleryId }) => {
   const [gallery, setGallery] = useState();
   const [images, setImages] = useState();
+  const { galleryName } = useParams();
 
   useEffect(() => {
-    Api.getGallery(galleryId)
+    Api.getGalleryByName(galleryName)
```

Comment:
Gallery names are not unique, what happens when two galleries have the same name?

Reply:
Good catch, this should be figured out, we can make it in the scope of the subdomain name. Or just use the gallery id